CONTRIBUTING.md

```diff
@@ -91,6 +118,38 @@ Once approved, the author of the pull request,
 or reviewer if the author does not have commit access,
 should "Squash and merge".
 
+### Design considerations
+
+#### Performance
+
+The agent is designed to monitor production applications. Therefore it's very important to keep the overhead of the agent as low as possible.
+
+It's not uncommon that you write or change code that can potentially change the performance characteristics of the agent and therefore also of the application's of our users.
+
+If this is the case then a perf. test should be added to the `test\Elastic.Apm.PerfTests` project which proves that the new code does not make the performance of the agent worse than it was before your PR.
+
+We care both about memory and CPU overhead and both should be measured. The `test\Elastic.Apm.PerfTests` is configured to measure both.
+
+#### Compatibility
+
+We aim to support the broadest set of .NET flavors and versions possible. This includes different OSs, typically Windows, Linux, and macOS.
+
+This is especially true for versions currently officially supported by Microsoft. In case of specific libraries and frameworks (e.g. ASP.NET Core) we aim to support every version which is supported by Microsoft or the library/framework author.
```

Comment:
"currently officially" sounds a little strange - how about "officially supported by Microsoft at the moment"

Reply:
Yeah, that sounds better, changed it.